src/cascadia/TerminalSettingsEditor/KeyChordListener.cpp

```diff
@@ -0,0 +1,147 @@
+                case VirtualKey::Control:
+                case VirtualKey::LeftControl:
+                case VirtualKey::RightControl:
+                {
+                    flags |= VirtualKeyModifiers::Control;
+                    break;
+                }
+                case VirtualKey::Menu:
+                case VirtualKey::LeftMenu:
+                case VirtualKey::RightMenu:
+                {
+                    flags |= VirtualKeyModifiers::Menu;
+                    break;
+                }
+                case VirtualKey::Shift:
+                case VirtualKey::LeftShift:
+                {
+                    flags |= VirtualKeyModifiers::Shift;
+                    break;
+                }
+                case VirtualKey::LeftWindows:
+                case VirtualKey::RightWindows:
+                {
+                    flags |= VirtualKeyModifiers::Windows;
+                    break;
+                }
+                }
+            }
+        }
+        return flags;
+    }
+
+    KeyChordListener::KeyChordListener()
+    {
+        InitializeComponent();
+        _InitializeProperties();
+
+        PropertyChanged([this](auto&&, const PropertyChangedEventArgs& args) {
+            if (args.PropertyName() == L"ProposedKeys")
+            {
+                KeyChordTextBox().Text(Model::KeyChordSerialization::ToString(_ProposedKeys));
+            }
+        });
+    }
+
+    void KeyChordListener::_InitializeProperties()
+    {
+        // Initialize any KeyChordListener dependency properties here.
+        // This performs a lazy load on these properties, instead of
+        // initializing them when the DLL loads.
+        if (!_CurrentKeysProperty)
+        {
+            _CurrentKeysProperty =
+                DependencyProperty::Register(
+                    L"CurrentKeys",
+                    xaml_typename<Control::KeyChord>(),
+                    xaml_typename<Editor::KeyChordListener>(),
+                    PropertyMetadata{ nullptr, PropertyChangedCallback{ &KeyChordListener::_OnCurrentKeysChanged } });
+        }
+    }
+
+    void KeyChordListener::_OnCurrentKeysChanged(DependencyObject const& d, DependencyPropertyChangedEventArgs const& e)
+    {
+        if (auto control{ d.try_as<Editor::KeyChordListener>() })
+        {
+            auto controlImpl{ get_self<KeyChordListener>(control) };
+            controlImpl->ProposedKeys(unbox_value<Control::KeyChord>(e.NewValue()));
+        }
+    }
+
+    void KeyChordListener::KeyChordTextBox_LosingFocus(IInspectable const& /*sender*/, LosingFocusEventArgs const& /*e*/)
+    {
+        // export our key chord into the attached view model
+        SetValue(_CurrentKeysProperty, _ProposedKeys);
+    }
+
+    void KeyChordListener::KeyChordTextBox_KeyDown(IInspectable const& /*sender*/, KeyRoutedEventArgs const& e)
+    {
+        const auto key{ e.OriginalKey() };
+        for (const auto mod : ModifierKeys)
+        {
+            if (key == mod)
```

Comment:
some implementations show "partial" keys as they go. like, if you press shift it'll say "shift+". Is that desirable here, or too complicated?

Reply:
I played with that a little bit. I ended up not liking it because plain modifiers aren't a valid key chord (at least for a key binding, not the object itself).

We _could_ change it at any time, but I vote we see how this feels in a selfhost/bug-bash, then iterate from there.